fitbenchmarking/core/fitting_benchmarking.py

```diff
@@ -553,18 +554,23 @@ def perform_fit(controller, options, grabbed_output):
     :type options: fitbenchmarking.utils.options.Options
     :param grabbed_output: Object that removes third part output from console
     :type grabbed_output: fitbenchmarking.utils.output_grabber.OutputGrabber
-    :return: The chi squared and runtime of the fit.
-    :rtype: tuple(float, float)
+    :return: The chi squared, runtime, and emissions of the fit.
+    :rtype: tuple(float, float, float)
     """
     num_runs = options.num_runs
+    controller.track_emissions = 'emissions' in options.table_type
+    emissions = np.inf
     try:
         with grabbed_output:
             controller.validate()
+            controller.prepare()
             # Calls timeit repeat with repeat = num_runs and number = 1
             runtime_list = timeit.Timer(
-                setup=controller.prepare,
                 stmt=controller.execute
             ).repeat(num_runs, 1)
+            if controller.track_emissions:
+                # stop emissions tracking after all runs have completed
+                emissions = controller.emissions_tracker.stop() / num_runs
```

Comment:
Good spot that we can just put prepare on the outside... I guess this means we can put the tracking back here instead of in the controller?

I'll put in a commit so you don't have to change it back again (if you think the commit shouldn't be there we can always git reset it away).